src/node/internal/internal_http_util.ts

```diff
@@ -24,3 +24,30 @@ export function once<RT>(
 
 export const kServerResponse = Symbol('ServerResponse');
 export const kIncomingMessage = Symbol('IncomingMessage');
+
+// RFC 7230 compliant header value splitting that respects quoted-string constructions
+// Spec: https://www.rfc-editor.org/rfc/rfc7230.html#appendix-B
+// Edge cases handled:
+// - Quoted strings with commas: 'text/plain; param="a, b"' -> 'text/plain; param="a, b"'
+// - Escaped quotes in strings: 'text/plain; param="a \"quoted\" value"' -> 'text/plain; param="a \"quoted\" value"'
+// - Multiple values: 'text/plain; f="a, b", text/html' -> 'text/plain; f="a, b"'
+// - Whitespace before commas: 'value1 \t , value2' -> 'value1'
+// - No commas: 'single-value' -> 'single-value'
```

Comment:
The comment here should indicate that only the first value up to the first comma is returned.